src/protocols/secure_channel/CASESession.cpp

```diff
@@ -646,6 +670,79 @@ CHIP_ERROR CASESession::RecoverInitiatorIpk()
     return CHIP_NO_ERROR;
 }
 
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+void CASESession::HandleConnectionComplete(Transport::ActiveConnectionState * conn, CHIP_ERROR err)
+{
+    char peerAddrBuf[chip::Transport::PeerAddress::kMaxToStringSize];
+    VerifyOrReturn(conn != nullptr);
+
+    Transport::PeerAddress peerAddr = conn->mPeerAddr;
+    peerAddr.ToString(peerAddrBuf);
+    CASESession * caseSession = reinterpret_cast<CASESession *>(conn->mAppState->appContext);
```

Comment:
What guarantees that conn->mAppState is still alive at this point?  These sorts of things need to be clearly documented.

Reply:
SessionManager checks for it. Added a VerifyOrDie for it to be non-null anyhow.